src/app/policies/components/PolicyConnections.vue

```diff
@@ -63,7 +63,16 @@ const props = defineProps<{
   mesh: string
   policyPath: string
   policyName: string
+  query: string
 }>()
 
-const searchInput = ref('')
+const emit = defineEmits<{
+  (event: 'query-change', query: string): void
+}>()
+
+const searchInput = ref(props.query)
+
+watch(() => searchInput.value, () => {
+  emit('query-change', searchInput.value)
+})
```

Comment:
Can we get rid of this component entirely and move its contents to the `*View`? It looks to be a single use component, plus its using DataSource internally which we should try to avoid. Then you shouldn't need all the extra plumbing.

I can't immediately see any reason why its its own component, but let me know if I've missed something.

Reply:
Whether a component is only used is not generally a good indicator for whether it should exist. A lot of our view components have grown in size and template logic to the point of obscurity and would benefit from smaller components for the sake of clarity.

In this case, it’s not much of a difference so I’m fine with the suggestions.

But generally speaking, we seemingly try to do everything in the view components and their templates nowadays and it makes things harder to understand. It promotes patterns like the `v-for` hack to introduce template scope variables that are better solved with separate components and computed properties.